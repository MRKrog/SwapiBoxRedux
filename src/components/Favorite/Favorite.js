import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

const Favorite = ({ favAmount, viewAllFavs }) => {
  return (
    <div className="Favorites">
      <NavLink to="/Favorites">
        <button onClick={viewAllFavs}>View Favorites<span className="count">{favAmount}</span></button>
      </NavLink>
    </div>
  )
}

Favorite.propTypes = {
  favAmount: PropTypes.number.isRequired,
  viewAllFavs: PropTypes.func.isRequired
}

export default Favorite;
