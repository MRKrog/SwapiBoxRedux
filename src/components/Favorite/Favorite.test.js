import React from 'react';
import { shallow } from 'enzyme';
import Favorite from './Favorite';

const totalFavs = 10;

describe('api', () => {
  let wrapper;
  let mockViewFavs;
  beforeEach(() => {
     mockViewFavs = jest.fn()
     
     wrapper = shallow(
       <Favorite favAmount={totalFavs}
                 viewAllFavs={mockViewFavs} />
     )
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should on click invoke the viewAllFavs function', () => {
    const button = wrapper.find('button');
    button.simulate('click');
    expect(mockViewFavs).toHaveBeenCalled()
  })

})
