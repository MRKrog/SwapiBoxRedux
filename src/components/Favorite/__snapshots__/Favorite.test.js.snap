// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`api should match the snapshot 1`] = `
<div
  className="Favorites"
>
  <NavLink
    to="/Favorites"
  >
    <button
      onClick={[MockFunction]}
    >
      View Favorites
      <span
        className="count"
      >
        10
      </span>
    </button>
  </NavLink>
</div>
`;
